### PR TITLE
Goinit

### PR DESCRIPTION
**Pull request by sebastianst**
Adds `tee` package with `Enclave` interface and core types.

### CODE REVIEW

File: tee/tee.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		// GenKeys generates a new secp256k1 ECDSA key and stores it as the
+		// enclave's signing key.
+		//
+		// It returns the public key derived Ethereum address and attestation of
+		// correct initialization of the enclave with the generated address. The
+		// attestation can be used to verify the enclave with the TEE vendor.
+		//
+		// The Operator must deploy the contract with the Enclave's address after
+		// calling GenKeys.
+		GenKeys() (common.Address, []byte, error)
```
Comment (RmbRT) [inline tee/tee.go]: I think that GenKey() or Setup() or Init() or something like that would be better here (as only one key is generated). I would also name the returned values OR make a type alias for the attestation.

File: tee/tee.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	Epoch = uint64
+
+	// A Block is a go-ethereum block together with its receipts. go-ethereum's
+	// types.Block type doesn't store the receipts...
+	Block struct {
+		types.Block
+		Receipts types.Receipts
+	}
+
+	Transaction struct {
```
Comment (RmbRT) [inline tee/tee.go:57]: Doesn't this need a sender and recipient?

File: tee/tee.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		epoch   Epoch
+		account common.Address
+		amount  *big.Int
+	}
+
+	Proof struct {
```
Comment (RmbRT) [inline tee/tee.go]: I think BalanceProof is clearer, but not absolutely necessary.

File: tee/tee.go (hunk 2 above)
Comment (RmbRT) [reply tee/tee.go:57]: and what about user signatures?

**Review by RmbRT — CHANGES_REQUESTED** [review]

File: tee/tee.go (hunk 1 above)
Comment (ndzik) [reply tee/tee.go]: Having it called `Init()` would also be more in line with other SGX implementations I saw.

Comment (sebastianst) [reply tee/tee.go]: Guys, this is a hackathon, lower your standards ;) Renamed to `Init`.

File: tee/tee.go (hunk 2 above)
Comment (sebastianst) [reply tee/tee.go:57]: You're totally right :joy:

**Pull request merged**